### PR TITLE
Add possibility to send header

### PR DESCRIPTION
**Pull request by pgt**
### Changelog

930b2a3 \* Add method `all` to entity
e5e0603 \* Add possibility to send header in request
### Motivation

Today we cannot send a header in a given request because likeno doesn't give this possibility. Also, we cannot get in the index route from a resource all the records. Here were added this possibility.

### CODE REVIEW

File: spec/likeno/entity_spec.rb [hunk 1]
```diff
@@ -201,9 +201,12 @@ class EntityTest < Likeno::Entity
   end
 
   describe 'find' do
+    let(:prefix) { '' }
```
Comment (marcheing) [inline spec/likeno/entity_spec.rb:204]: Why this change? Why not just add these parameter on the next context? These are the default values for these parameters, therefore a perfectly reasonable use for the method is to not explicitly include them on the method call and expect to function exactly as before.

File: lib/likeno/entity.rb [hunk 2]
```diff
@@ -95,11 +95,20 @@ def self.exists?(id)
...
     end
 
-    def self.find(id)
-      response = request(find_action, id_params(id), :get)
+    def self.find(id, prefix = '', headers = {})
+      response = request(find_action, id_params(id), :get, prefix, headers)
       new(response[entity_name], true)
     end
 
+    def self.all(headers = {}, params = {})
```
Comment (marcheing) [inline lib/likeno/entity.rb:103]: Wouldn't it be better to add an `all_action` on the [helper](https://github.com/pgt/likeno/blob/add-possibility-to-send-header/lib/likeno/helpers/crud_request_parameters.rb)? The action may not always be an empty string.

Also, the prefix may not always be empty as well. I can agree the method is most certainly going to be `get` though.

**Comment by rafamanzo** [comment]
Thank you for the contributions!

We've split this into other three smaller PRs:
- **header support**: https://github.com/mezuro/likeno/pull/12
- **prefix and header for `find` and `exists`**: https://github.com/mezuro/likeno/pull/13
- **add `all` method**: https://github.com/mezuro/likeno/pull/14

Please notice that #13 and #14 changes the `header` and `prefix` from method attributes to methods following the same standards from `CRUDRequestParameters` module. So, if you want to send custom `headers`, `prefix`, `params` or `action` you need to override the respective methods on `Entity`'s subclasses.

Finally on #12 have a look at the spec there. We've fixed it to fail if the request has no headers.

Thank you again :trophy:
